hooks/test/browser/useMemo.test.js

```diff
@@ -42,4 +41,40 @@ describe('useMemo', () => {
 		expect(memoFunction).to.have.been.calledTwice;
 	});
 
+	it('short circuits diffing for memoized components', () => {
+		let spy = sinon.spy();
+		let spy2 = sinon.spy();
+		const X = ({ count }) => {
+			spy();
+			return <span>{count}</span>;
+		};
+
+		const Y = ({ count }) => {
+			spy2();
+			return <p>{count}</p>;
+		};
+
+		const App = ({ x }) => {
+			const y = useMemo(() => <Y count={x} />, [x]);
+			return (
+				<div>
+					<X count={x} />
+					{y}
+				</div>
+			);
+		};
+
+		render(<App x={0} />, scratch);
+		expect(spy).to.be.calledOnce;
+		expect(spy2).to.be.calledOnce;
+		expect(scratch.innerHTML).to.equal('<div><span>0</span><p>0</p></div>');
+		render(<App x={0} />, scratch);
```

Comment:
super tiny nit: Personally I prefer a blank line before the next update so that it's easier to scan for the eyes to see what's happening :)